synapse_invite_checker/invite_checker.py

```diff
@@ -275,12 +275,12 @@ async def _raw_federation_list_fetch(self) -> str:
         return resp.decode()
 
     async def _raw_gematik_root_ca_fetch(self) -> dict:
-        return await self.api.http_client.get_json(
+        return await self.api._hs.get_proxied_http_client().get_json(
```

Comment:
We should at least leave a comment explaining why we're accessing a private member here; maybe we can get this to become a public interface upstream longer-term?

Reply:
As mentioned by emily, we apparently access this all over the place already, so that ship has sailed.

Would still be nice to see if we can get the things we need to become public long-term, but that's beyond the scope of this PR. Let's move this to a separate issue?